### PR TITLE
Introduction

### PR DESCRIPTION
**Pull request by gdozot2**

### CODE REVIEW

File: src/components/FullImage/FullImage.vue [hunk 1]
```diff
@@ -0,0 +1,250 @@
+<template>
+  <div v-if="this.imageData" class="information">
```
Comment (slecorne) [inline src/components/FullImage/FullImage.vue:2]: you don't need `this.` in templates, remove them to make it easier to read

File: src/components/FullImage/FullImage.vue [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+      imageData: undefined,
+      collection: undefined,
+      viewer: undefined,
+    };
+  },
+  methods: {
+    previousImage() {
+      if (this.currentImage > 0) {
+        this.currentImage = this.currentImage - 1;
+        this.imageData[0] = this.collection[this.currentImage];
```
Comment (slecorne) [inline src/components/FullImage/FullImage.vue:75]: does not need to be an array, it sound sover complicated to do `this.imageData[0]` all the time, just do `this.imageData = `

File: src/components/FullImage/FullImage.vue [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      vm.imageId = to.params.index;
+    });
+  },
+  data() {
+    return {
+      imageId: undefined,
+      currentImage: undefined,
+      imageData: undefined,
+      collection: undefined,
+      viewer: undefined,
```
Comment (slecorne) [inline src/components/FullImage/FullImage.vue:68]: please add comments and maybe rename some vars, what is currentImage (index), what is a collection?

File: src/components/History/History.vue [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+      forthImage: undefined,
+      forthPosition: 0,
+      currentHistoryLength: 0,
+      position: 0,
+    };
+  },
+  watch: {
+    history: {
+      handler(historyElement) {
+        if (historyElement.length > 3) {
```
Comment (slecorne) [inline src/components/History/History.vue:165]: please comment, hard to follow

File: src/components/History/History.vue [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+          if (this.currentHistoryLength !== historyElement.length) {
+            if (this.currentHistory.length % 4 === 1) {
+              setTimeout(() => (this.secondImage = undefined), 1000);
```
Comment (slecorne) [inline src/components/History/History.vue:168]: why setTimeout?

File: src/components/ImageSelector/ImageSelector.vue [hunk 6]
```diff
@@ -45,57 +275,537 @@
...
+              decade: this.currentDecade,
+              index: this.currentIndex,
+              data: this.data[this.currentIndex].imagePaths.square,
+            });
+          }, 8000)
+        );
+      }
+    },
+    // Watcher uses to manage the completion element
+    currentDecade: function() {
```
Comment (slecorne) [inline src/components/ImageSelector/ImageSelector.vue:527]: probably you could use a computed property instead of watch

File: src/components/ImageSelector/ImageSelector.vue [hunk 7]
```diff
@@ -422,31 +1733,61 @@ export default {
...
+    }
   },
 };
 </script>
 
 <style scoped>
 @import "./imageselector.css";
 @import "./sliderspeed.css";
+@import "./carouselAnimation.css";
 </style>
```
Comment (slecorne) [inline src/components/ImageSelector/ImageSelector.vue:1793]: this file is probably too big and should ideally either be splitted in different components (completion spiral could be I guess a specific vue component to reuse everywhere).

Also would need some refactoring, hard to follow and probably hard to debug for the different position computations with multiple parameters.
I would not do it now since too risky but to keep in mind if we work again on that

File: src/components/ImageSelector/carouselAnimation.css [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+    mask-image: radial-gradient(
+      circle at 50% 50%,
+      black 10%,
+      rgba(0, 0, 0, 0.1) 91%
+    );
+  }
+  90% {
+    mask-image: radial-gradient(
+      circle at 50% 50%,
+      black 10%,
```
Comment (slecorne) [inline src/components/ImageSelector/carouselAnimation.css:447]: can you check to not specifiy again all the value that do not change (I guess it works), would make it much lighter to read and understand the animations

File: src/components/ImageSelector/relatedImageAnimation.css [hunk 9]
```diff
@@ -0,0 +1,83 @@
+.relatedImageBase {
+    animation: relatedImageBaseAnimation 3s linear 0s normal forwards;
+}
+
+@keyframes relatedImageBaseAnimation {
+    0% {
```
Comment (slecorne) [inline src/components/ImageSelector/relatedImageAnimation.css:6]: could you specify only 0% and 100% ? It should animate automatically between the 2

File: src/components/ImageSelector/relatedImageAnimation.css [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+.removeRelatedImageBaseText {
+  animation: removeRelatedImageBaseAnimation 3s linear 0s normal forwards;
+}
+
+.removeRelatedImageBase {
+  animation: removeRelatedImageBaseAnimation 8s linear 0s normal forwards;
+}
+
+@keyframes removeRelatedImageBaseAnimation {
+  0% {
```
Comment (slecorne) [inline src/components/ImageSelector/relatedImageAnimation.css:50]: same here

**Review by slecorne — CHANGES_REQUESTED** [review]

File: src/api/dataFetching.js [hunk 11]
```diff
@@ -2,41 +2,162 @@ import axios from "axios";
...
+    };
+
+    const { headers } = await axios.get(process.env.VUE_APP_FETCH_BASE, params);
+
+    const totalImages = headers["omeka-s-total-results"];
+
+    params.params["per_page"] = 1;
+    params.params["page"] = Math.floor(Math.random() * (totalImages));
+
+    const { data } = await axios.get(process.env.VUE_APP_FETCH_BASE, params);
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/api/dataFetching.js:132]: Unfortunately we cannot do that. Without specification, omeka s returns 25 elements per page. But the max return elements is 1000. So this code could work but if we have less than 1000 elements.

[the remark this answers is not in the document]
Comment (slecorne) [reply src/api/dataFetching.js:132]: ok, makes sense

File: src/components/History/History.vue (hunk 5 above)
Comment (gdozot2) [reply src/components/History/History.vue:168]: The timeout is used in this method for animation purpose. The image appears or disappears only when the left move animation is done.

File: src/api/dataFetching.js [hunk 12]
```diff
@@ -2,41 +2,162 @@ import axios from "axios";
...
+  let title = element["dcterms:title"];
+  let description = element["dcterms:description"] ? element["dcterms:description"][0]["@value"] : null
+  if (title) {
+    return new ImageData(
+      element["dcterms:identifier"][0]["@value"],
+      title[0]["@value"],
+      element["dcterms:creator"][0]["@value"],
+      element["dcterms:medium"][0]["@value"],
+      element["thumbnail_display_urls"],
+      element["dcterms:coverage"],
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/api/dataFetching.js:15]: Ok, when the indian finish the job, we backup the DB then I switch that

File: src/components/ImageSelector/ImageSelector.vue (hunk 7 above)
Comment (gdozot2) [reply src/components/ImageSelector/ImageSelector.vue:1793]: Yes, I think too that a refactoring is necessary

File: src/api/dataFetching.js [hunk 13]
```diff
@@ -2,41 +2,162 @@ import axios from "axios";
...
+      null,
+      element["dcterms:creator"][0]["@value"],
+      element["dcterms:medium"][0]["@value"],
+      element["thumbnail_display_urls"],
+      element["dcterms:coverage"],
+      element["dcterms:created"][0]["@value"],
+      element["o:media"][0]["@id"],
+      description
+    );
+  }
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/api/dataFetching.js]: Done

File: src/api/dataFetching.js [hunk 14]
```diff
@@ -2,41 +2,162 @@ import axios from "axios";
...
+    };
+    const { data } = await axios.get(process.env.VUE_APP_FETCH_BASE, params);
+    return parseImages(data);
+  },
+
+  // Voir si en utilisant le full text search avec l'id c'est mieux
+  async getRelatedImages(tag, id) {
+    const params = {
+      params: {
+        "fulltext_search": tag["@value"],
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/api/dataFetching.js]: Done and test

File: src/api/dataFetching.js [hunk 15]
```diff
@@ -2,41 +2,162 @@ import axios from "axios";
...
+    params.params["page"] = Math.floor(Math.random() * (totalImages));
+
+    const { data } = await axios.get(process.env.VUE_APP_FETCH_BASE, params);
+    if (data.length > 0) {
+      return parseElement(data[0]);
+    }
+    return undefined;
+  },
+
+  async getImageByTitle(title) {
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/api/dataFetching.js]: Done

File: src/components/ImageSelector/relatedImageAnimation.css (hunk 9 above)
Comment (gdozot2) [reply src/components/ImageSelector/relatedImageAnimation.css]: Right, modification done

File: src/components/ImageSelector/relatedImageAnimation.css (hunk 10 above)
Comment (gdozot2) [reply src/components/ImageSelector/relatedImageAnimation.css:23]: Right too, modification done

File: src/components/FullImage/ArrowLeft.vue [hunk 16]
```diff
@@ -0,0 +1,21 @@
+<template>
```
[the remark this answers is not in the document]
Comment (gdozot2) [reply src/components/FullImage/ArrowLeft.vue]: Right, replacement done

File: src/components/ImageSelector/ImageSelector.vue (hunk 6 above)
Comment (gdozot2) [reply src/components/ImageSelector/ImageSelector.vue:527]: Probably yes, but this watch manage some par tof the completion element, I don't know if it's a good idea to change it now. Maybe, that could be better during the refactoring of the component. What do you think?

File: src/components/ImageSelector/carouselAnimation.css (hunk 8 above)
Comment (gdozot2) [reply src/components/ImageSelector/carouselAnimation.css:447]: Unfortunately, seems that not works. radial gradient function needs the informations to compute the gradient. I search and try but that doesn't work.

File: src/components/ImageSelector/ImageSelector.vue (hunk 6 above)
Comment (slecorne) [reply src/components/ImageSelector/ImageSelector.vue:527]: probably best to refactor later indeed

**Review by slecorne — APPROVED** [review]
